aten/src/ATen/native/TensorIterator.h

```diff
@@ -85,7 +86,21 @@ struct CAFFE2_API OperandInfo {
   /// input should be converted to this type if necessary. For outputs, this
   /// specifies which type to allocate. Note that there is very limited support
   /// for type conversions currently: they are only allowed for zero-dim tensors.
-  Type* type = nullptr;
+  Backend backend = Backend::Undefined;
```

Comment:
Are you sure you want Backend here and not DeviceType? DeviceType seems better to me, since TensorIterator is probably not going to work on a SparseCPU tensor...